### PR TITLE
Victor K express-notes-api (1st submission)

### PR DESCRIPTION
**Pull request by vkalinkin**
notes api 1
notes api 2
notes api 3
notes api 4
notes api 5
notes api 6
notes api 7

### CODE REVIEW

File: express-notes-api/index.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    res.status(400).send('"error": "id must be a positive integer"');
+  } else {
+    for (const property in notesData) {
+      if (property === currentId) {
+        foundId = true;
+        res.json(notesData[property]);
+        fs.writeFile('data.json');
+        break;
+      }
+    }
```
Comment (scbowler) [inline express-notes-api/index.js:33]: You don't need a loop to find a value in an object if you know the property name. You can just do:

```JAVASCRIPT
if(!notesData[currentId]) {
  return res.status(404).json({error: `cannot find note with id ${currentId}`});
}
```

File: express-notes-api/index.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  const integerId = parseInt(currentId);
+
+  if (integerId < 0 || isNaN(integerId)) {
+    res.status(400).send('"error": "id must be a positive integer"');
+  } else {
+    for (const property in notesData) {
+      if (property === currentId) {
+        foundId = true;
+        res.json(notesData[property]);
+        fs.writeFile('data.json');
```
Comment (scbowler) [inline express-notes-api/index.js:30]: There was no changes to the data so you don't need to `writeFile` here

File: express-notes-api/index.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        break;
+      }
+    }
+    if (foundId === false) {
+      res.status(404).send(`"error": "cannot find note with id ${currentId}"`);
```
Comment (scbowler) [inline express-notes-api/index.js:35]: We want to use `.json()` over `.send()`. Also make sure the data you are sending is an object.

File: express-notes-api/index.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    }
+  }
+});
+
+app.post('/api/notes', function (req, res) {
+  const message = req.body;
+  const msg = message.content;
+  if (msg === undefined) {
```
Comment (scbowler) [inline express-notes-api/index.js:43]: `msg` might also be an empty string so here is would be better to just check if `msg` is falsey

```JAVASCRIPT
if (!msg) {
```

File: express-notes-api/index.js [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+        const newData = JSON.stringify(notes, null, 2);
+        fs.writeFile('data.json', newData, 'utf8', err => {
+          if (err) {
+            return res.status(500).send('"error": "An unexpected error occurred."');
+          }
+          res.sendStatus(204);
+        });
+        break;
+      }
+    }
```
Comment (scbowler) [inline express-notes-api/index.js:86]: Same here you don't need a loop, you can just directly check if the value exists or not:

```JAVASCRIPT
if(noteData[currentId]) {
  // delete note and update file
} else {
  // Send 404 error
}
```

File: express-notes-api/index.js [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+  const message = req.body;
+  const msg = message.content;
+  let foundId = false;
+
+  if (integerId < 0 || isNaN(integerId)) {
+    res.status(400).send('"error": "id must be a positive integer"');
+  } else if (msg === undefined) {
+    res.status(400).send('"error": "content is a required field"');
+  } else {
+    for (const property in notesData) {
```
Comment (scbowler) [inline express-notes-api/index.js:106]: Same here, don't need a loop

**Review by scbowler — CHANGES_REQUESTED** [review]
Looking good, just needs a bit of cleanup. Let me know if you have any questions.

File: express-notes-api/index.js (hunk 3 above)
Comment (vkalinkin) [reply express-notes-api/index.js:35]: So I should format the error message as a json object?

**Pull request merged**